presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/JtsGeometryUtils.java

```diff
@@ -128,12 +190,28 @@ private static Point readPoint(SliceInput input)
         return GEOMETRY_FACTORY.createPoint(coordinates);
     }
 
+    private static void writePoint(Point point, DynamicSliceOutput sliceOutput) {
+        requireNonNull(sliceOutput, "output is null");
+        if (!point.isEmpty()) {
+            sliceOutput.writeInt(EsriShapeType.POINT.code);
+            writeCoordinate(point.getCoordinate(), sliceOutput);
+        } else {
+            throw new IllegalArgumentException("Trying to write an empty point.");
```

Comment:
Empty geometries are valid objects.

Reply:
Could a point geometry be empty (meaning, its type is Point, yet have no coordinates?)
If so, what should I put for x and y? Would 0 do?

Reply:
Any geometry type can be empty, including points. "POINT EMPTY" WKT represents an empty point. To represent empty point, use NaN (see isEsriNaN function call in readPoint).